doc/devel/coding_guide.rst

```diff
@@ -1,369 +1,318 @@
-  not part of other non-stylistic work because it obscures the git history of
-  functional changes to the code. Reflowing a method or docstring as part of a
-  larger refactor/rewrite is acceptable.
+Import the following modules using the standard scipy conventions::
 
+  import numpy as np
+  import numpy.ma as ma
+  import matplotlib as mpl
+  import matplotlib.pyplot as plt
+  import matplotlib.cbook as cbook
+  import matplotlib.patches as mpatches
 
-Organizational topics:
+In general, Matplotlib modules should **not** import `.rcParams` using ``from
+matplotlib import rcParams``, but rather access it as ``mpl.rcParams``.  This
+is because some modules are imported very early, before the `.rcParams`
+singleton is constructed.
 
-.. rst-class:: checklist
+Variable names
+==============
 
-* Make sure all :ref:`automated tests <pr-automated-tests>` pass.
-* The PR should :ref:`target the main branch <pr-branch-selection>`.
-* Tag with descriptive :ref:`labels <pr-labels>`.
-* Set the :ref:`milestone <pr-milestones>`.
-* Keep an eye on the :ref:`number of commits <pr-squashing>`.
-* Approve if all of the above topics are handled.
-* :ref:`Merge  <pr-merging>` if a sufficient number of approvals is reached.
+When feasible, please use our internal variable naming convention for objects
+of a given class and objects of any child class:
 
-.. _pr-guidelines-details:
++------------------------------------+---------------+------------------------------------------+
+|             base class             | variable      |                multiples                 |
++====================================+===============+==========================================+
+| `~matplotlib.figure.FigureBase`    | ``fig``       |                                          |
++------------------------------------+---------------+------------------------------------------+
+| `~matplotlib.axes.Axes`            | ``ax``        |                                          |
++------------------------------------+---------------+------------------------------------------+
+| `~matplotlib.transforms.Transform` | ``trans``     | ``trans_<source>_<target>``              |
++                                    +               +                                          +
+|                                    |               | ``trans_<source>`` when target is screen |
++------------------------------------+---------------+------------------------------------------+
 
-Detailed guidelines
-===================
+Generally, denote more than one instance of the same class by adding suffixes to
+the variable names. If a format isn't specified in the table, use numbers or
+letters as appropriate.
 
-.. _pr-documentation:
+API changes
+===========
 
-Documentation
--------------
+If you are adding new features, changing behavior or function signatures, or
+removing classes, functions, methods, or properties, please follow these
+guidelines:
+
+.. toctree::
```

Comment:
I'm slightly uneasy with burrying toctrees deep inside page content. This makes it difficult to understand the toc structure.

It would be better to collect the toc in one central place (or at least at the top of the respective page. This is possible by making it hidden. And then use regular links here. - Which I also think can be better integrated. The rendered toc looks a bit out-of-place here.